File: app/src/main/java/gowtham/com/projectcloud/dagger/WeatherApplication.java
package gowtham.com.projectcloud.dagger;

import android.app.Application;

public class WeatherApplication extends Application {

    private ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        component = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();
    }
}

File: app/src/main/java/gowtham/com/projectcloud/dagger/WeatherApplication.kt
package gowtham.com.projectcloud.dagger

import android.app.Application
import gowtham.com.projectcloud.modules.NetworkModule

class WeatherApplication : Application() {

    var component: ApplicationComponent? = null

    override fun onCreate() {
        super.onCreate()

        component = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .networkModule(NetworkModule(this))
                .build()
    }
}
